src/main/java/org/openrewrite/java/testing/cleanup/RemoveVisibleForTestingAnnotationWhenUsedInProduction.java

```diff
@@ -0,0 +1,176 @@
+
+    public static class Scanned {
+        List<String> methods = new ArrayList<>();
+        List<String> fields = new ArrayList<>();
+        List<String> classes = new ArrayList<>();
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Remove `@VisibleForTesting` annotation when target is used in production";
+    }
+
+    @Override
+    public String getDescription() {
+        return "The `@VisibleForTesting` annotation is used when a method or a field has been made more accessible then it would normally be, solely for testing purposes. " +
+                "This recipe removes the annotation where such an element is used from production classes. It identifies production classes as classes in `src/main` and test classes as classes in `src/test`. " +
+                "It will remove the `@VisibleForTesting` from methods, fields (both member fields and constants), constructors and inner classes. " +
+                "This recipe should not be used in an environment where QA tooling acts on the `@VisibleForTesting` annotation.";
+    }
+
+    @Override
+    public Scanned getInitialValue(ExecutionContext ctx) {
+        return new Scanned();
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getScanner(Scanned acc) {
+        return new JavaIsoVisitor<ExecutionContext>() {
+
+            @Override
+            public J.FieldAccess visitFieldAccess(J.FieldAccess fieldAccess, ExecutionContext ctx) {
+                // Mark classes
+                if (fieldAccess.getTarget().getType() instanceof JavaType.Class) {
+                    checkAndRegister(acc.classes, fieldAccess.getTarget().getType());
+                }
+                // Mark fields
+                if (fieldAccess.getName().getFieldType() != null) {
+                    checkAndRegister(acc.fields, fieldAccess.getName().getFieldType());
+                }
+                return super.visitFieldAccess(fieldAccess, ctx);
+            }
+
+            @Override
+            public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
+                if (method.getMethodType() != null) {
+                    checkAndRegister(acc.methods, method.getMethodType());
+                }
+                return super.visitMethodInvocation(method, ctx);
+            }
+
+            @Override
+            public J.NewClass visitNewClass(J.NewClass newClass, ExecutionContext ctx) {
+                // Mark constructors
+                if (newClass.getConstructorType() != null) {
+                    checkAndRegister(acc.methods, newClass.getConstructorType());
+                }
+                // Mark classes
+                if (newClass.getClazz() != null && newClass.getClazz().getType() != null && newClass.getClazz().getType() instanceof JavaType.Class) {
```

Comment:
```suggestion
                if (newClass.getClazz() != null && newClass.getClazz().getType() instanceof JavaType.Class) {
```

The extra null check is unnecessary. An `instanceof` check is _also_ a null check.